File: 3.py
# Problem 3 
# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143 ?

import math

def isPrime(num):
  if num < 2:
    return False
  i = 2
  while i <= math.sqrt(num):
    if num % i == 0:
      return False
    i += 1
  return True

num = 600851475143
max_prime = 2

while num != 1:
  #print num
  if isPrime(num): # If the number is already prime, then break out of loop
    if num > max_prime:
      max_prime = num
    break
  else:
    i = 2
    while i < num:
      if isPrime(i) and num % i == 0:
        num = num / i        
        if i > max_prime:
          max_prime = i
      i += 1


print max_prime

File: C/3.c
#include<stdio.h>
#include<math.h>

/*
 * isPrime - Function to check whether a number is a prime number
 *
 * @ret:
 * 1 if Prime. 0 if not Prime. 
 *
 * @param:
 * int num - Number to be checked
*/
int isPrime(double num)
{
	double i = 0.00;
	if (num < 2)
	{
		return 0;
	} else {
		while (i < (sqrt(num)))
		{
			if ((int)fmod(num, i) == 0)
			{
				return 0;
			}
			i = i + 1;
		}
		return 1;
	}
}

int main(void)
{
	double i = 0;
	double largetPrime = 0;
	double number = 600851475143;

	while (i < number) {
		//printf("Checking %lf...\n", i);
		if ((int)fmod(number, i) == 0)
		{
			if (isPrime(i))
				largetPrime = i;
		}
		i = i + 1;
	}

	printf("Largest prime factor of %lf is %lf\n", number, largetPrime);
	return 0;
}